apps/server/src/modules/oauth-provider/uc/oauth-provider.login-flow.uc.ts

```diff
@@ -44,21 +48,57 @@ export class OauthProviderLoginFlowUc {
 		loginRequestBody: LoginRequestBody
 	): Promise<ProviderRedirectResponse> {
 		const loginResponse: ProviderLoginResponse = await this.oauthProviderService.getLoginRequest(challenge);
-		const pseudonym: PseudonymDO = await this.oauthProviderLoginFlowService.getPseudonym(currentUserId, loginResponse);
-		const acceptLoginRequestBody: AcceptLoginRequestBody =
-			this.oauthProviderRequestMapper.mapCreateAcceptLoginRequestBody(
-				loginRequestBody,
-				currentUserId,
-				pseudonym.pseudonym
-			);
-		await this.oauthProviderLoginFlowService.validateNextcloudPermission(currentUserId, loginResponse);
+
+		if (!loginResponse.client.client_id) {
+			throw new InternalServerErrorException(`Cannot find oAuthClientId in login response for challenge: ${challenge}`);
+		}
+
+		const tool: ExternalToolDO | LtiToolDO = await this.oauthProviderLoginFlowService.findToolByClientId(
+			loginResponse.client.client_id
+		);
+
+		if (!tool.id) {
+			throw new InternalServerErrorException('Tool has no id');
+		}
+
+		if (this.oauthProviderLoginFlowService.isNextcloudTool(tool)) {
```

Comment:
Maybe we can talk with alex to extend the ctl model for required permissions to launch